diff --git a/common.gypi b/common.gypi
index de83a56672..1f5c1781bb 100644
--- a/common.gypi
+++ b/common.gypi
@@ -23,7 +23,7 @@
     'node_shared_openssl%': 'false',

     'node_tag%': '',
-    'uv_library%': 'static_library',
+    'uv_library%': 'shared_library',

     'clang%': 0,
     'error_on_warn%': 'false',
diff --git a/deps/uv/uv.gyp b/deps/uv/uv.gyp
index 6c86c3fa50..006596e86a 100644
--- a/deps/uv/uv.gyp
+++ b/deps/uv/uv.gyp
@@ -151,7 +151,21 @@
     ],
   },

-  'targets': [
+  'targets': [  {
+    'target_name': 'libuv_postbuild',
+    'type': 'none',
+    'dependencies': ['libuv'],
+    'conditions': [
+      ['OS=="win"', {
+        'copies': [{
+          'destination': '<(PRODUCT_DIR)/lib',
+          'files': [
+              '<(PRODUCT_DIR)/libuv<(STATIC_LIB_SUFFIX)',
+          ],
+        }],
+      }],
+    ],
+  },
     {
       'target_name': 'libuv',
       'toolsets': ['host', 'target'],
@@ -247,7 +261,7 @@
               # This will cause gyp to set soname
               # Must correspond with UV_VERSION_MAJOR
               # in include/uv/version.h
-              'product_extension': 'so.1',
+              'product_extension': 'so',
             }],
           ],
         }],
diff --git a/node.gyp b/node.gyp
index 11474953b1..a45c909938 100644
--- a/node.gyp
+++ b/node.gyp
@@ -19,7 +19,7 @@
     'node_shared_zlib%': 'false',
     'node_shared_http_parser%': 'false',
     'node_shared_cares%': 'false',
-    'node_shared_libuv%': 'false',
+    'node_shared_libuv%': 'true',
     'node_shared_sqlite%': 'false',
     'node_shared_uvwasi%': 'false',
     'node_shared_nghttp2%': 'false',
@@ -860,6 +860,7 @@
         'deps/histogram/histogram.gyp:histogram',
         'deps/nbytes/nbytes.gyp:nbytes',
         'node_js2c#host',
+        'deps/uv/uv.gyp:libuv_postbuild',
       ],

       'sources': [
diff --git a/node.gypi b/node.gypi
index 44780fe3b1..10346730bf 100644
--- a/node.gypi
+++ b/node.gypi
@@ -188,13 +188,6 @@
               '-Wl,-force_load,<(PRODUCT_DIR)/libuv<(STATIC_LIB_SUFFIX)',
             ],
           },
-          'msvs_settings': {
-            'VCLinkerTool': {
-              'AdditionalOptions': [
-                '/WHOLEARCHIVE:<(PRODUCT_DIR)/lib/libuv<(STATIC_LIB_SUFFIX)',
-              ],
-            },
-          },
           'conditions': [
             ['OS!="aix" and OS!="os400" and OS!="ios" and node_shared=="false"', {
               'ldflags': [
