diff --git forkSrcPrefix/configure.py forkDstPrefix/configure.py
index 9c6c0a56ccac6d29d1085bcdf40740b7a85c584c..18dfea6f76ef3383ff44c905f62863c56612c5fe 100755
--- forkSrcPrefix/configure.py
+++ forkDstPrefix/configure.py
@@ -1573,7 +1573,7 @@ def configure_node(o):
   elif sys.platform.startswith('zos'):
     shlib_suffix = '%s.x'
   else:
-    shlib_suffix = 'so.%s'
+    shlib_suffix = 'so'
   if '%s' in shlib_suffix:
     shlib_suffix %= node_module_version
 
